test/integration/shell_int_test.go

```diff
@@ -504,6 +504,28 @@ func (suite *ShellIntegrationTestSuite) TestWindowsShells() {
 	cp.ExpectExitCode(0)
 }
 
+func (suite *ShellIntegrationTestSuite) TestAsync() {
```

Comment:
I'm wary of adding a test like this because it'll take a few minutes to run (particularly on Windows), but I don't see another way to test this without enabling `optin.unstable.async_runtime` for all tests. Doing so might be very disruptive as we currently have an "opt-in" mechanic for async runtimes.